pvlib/ivtools.py

```diff
@@ -262,6 +262,173 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
+                   cells_in_series, EgRef=1.121, dEgdT=-0.0002677,
+                   temp_ref=25, irrad_ref=1000):
+    """
+    Calculates the parameters for the De Soto single diode model using the
+    procedure described in [1]. This procedure has the advantage of
+    using common specifications given by manufacturers in the
+    datasheets of PV modules. The solution is found using the
+    scipy.optimize.root() function, with the correpsonding default solver
+    method 'hybr'.
+    The parameters returned by this function can be used by
+    pvsystem.calcparams_desoto to calculate the values at different
+    irradiance and cell temperature.
+
+    Parameters
+    ----------
+     v_mp: float
+        Module voltage at the maximum-power point at reference conditions [V].
+    i_mp: float
+        Module current at the maximum-power point at reference conditions [A].
+    v_oc: float
+        Open-circuit voltage at reference conditions [V].
+    i_sc: float
+        Short-circuit current at reference conditions [A].
+    alpha_sc: float
+        The short-circuit current (i_sc) temperature coefficient of the
+        module [A/K].
+    beta_voc: float
+        The open-circuit voltage (v_oc) temperature coefficient of the
+        module [V/K].
+    cells_in_series: float
+        Number of cell in the module.
+    EgRef: float, default 1.121 eV - value for silicon
+        Energy of bandgap of semi-conductor used [eV]
+    dEgdT: float, default -0.0002677 - value for silicon
+        Variation of bandgap according to temperature [eV/K]
+    temp_ref: float, default 25
+        Reference temperature condition [C]
+    irrad_ref: float, default 1000
+        Reference irradiance condition [W/m2]
+
+    Returns
+    -------
+    Dictionary with the following elements:
+
+    * I_L_ref: float
+        Light-generated current at reference conditions [A]
+    * I_o_ref: float
+        Diode saturation current at reference conditions [A]
+    * R_s: float
+        Series resistance [ohms]
+        Note that '_ref' is not mentionned in the name because
+        this resistance is not sensible to the conditions of test.
+    * R_sh_ref: float
+        Shunt resistance at reference conditions [ohms].
+    * a_ref: float
+        Modified ideality factor at reference conditions.
+        The product of the usual diode ideality factor (n, unitless),
+        number of cells in series (Ns), and cell thermal voltage at
+        specified effective irradiance and cell temperature.
+    * alpha_sc: float
+        The short-circuit current (i_sc) temperature coefficient of the
+        module [A/K].
+    * EgRef: float
+        Energy of bandgap of semi-conductor used [eV]
+    * dEgdT: float
+        Variation of bandgap according to temperature [eV/K]
+    * irrad_ref: float
+        Reference irradiance condition [W/m2]
+    * temp_ref: float
+        Reference temperature condition [C]
+    * result: scipy.optimize.OptimizeResult
```

Comment:
Key is actually named `optimize_result`. Don't lump this in with the rest of parameters, see comment below.